docker_101/docker_101.md

```diff
@@ -0,0 +1,395 @@
+
+long_description: If you've been curious about how to use docker for your research, this module is a great place to start. The Docker 101 tutorial is a popular, hands-on approach to learning docker that will get you using containers right away, so you can learn by doing. To help you bridge the gap between basic docker use and best practices for using docker in research, we also link to an article outlining 10 rules to help you create great containers for research, and lists of ready-to-use docker images for a variety of analysis workflows. This module includes running and editing comamnds in the terminal, so you'll need some basic familiarity with bash, but it is otherwise appropriate for beginniners. No prior experience with docker or containers is assumed. 
+
+estimated_time_in_minutes: 60
+
+@pre_reqs
+This module assumes no prior experience with containers, and no particular coding other than some familiarity with the command line, such as being able to change directories and run bash commands that will be supplied for you to copy and paste. You will need to create and edit text files in a text editor like VSCode. 
+
+You'll also need to create an account on [Docker Hub](https://hub.docker.com/) (it's free), if you don't have one already, and you'll need to be able to install the Docker Desktop software on your machine (also free). 
+@end
+
+@learning_objectives  
+After completion of this module, learners will be able to:
+
+- Use the command line to create and run a container from a Dockerfile
+- Share containers 
+- Understand both technical requirements and best practices for writing Dockerfiles for use in research
+@end
+
+resource1_name: Docker 101 Tutorial
+resource1_description: From the Docker website: "In this self-paced, hands-on tutorial, you will learn how to build images, run containers, use volumes to persist data and mount in source code, and define your application using Docker Compose. You’ll even learn about a few advanced topics, such as networking and image building best practices."
+resource1_wellvetted: true
+resource1_wellvetted_text:  The Docker 101 tutorial is an [open source project](https://github.com/docker/getting-started), so it has many authors, but it is sponsored and hosted by Docker Inc, the company that produces the docker software. The open source nature of the tutorial also helps ensure that any errors or problems can be caught and addressed quickly.
+resource1_maintained: true
+resource1_maintained_text: This tutorial is maintained by Docker Inc, so we expect that it will remain up to date whenever changes are implemented in docker.
+resource1_stablesupport: true
+resource1_stablesupport_text: This is hosted on the Docker website, and it is a popular and widely-shared tutorial. We expect it will continue to be available for the foreseeable future.
+resource1_a11y_issues: No known issues with accessibility, but we may have missed something. If you encounter an issue, please [let us know](#feedback)!
```

Comment:
It isn't actually supposed to, for the a11y issues. Originally that was the thought, but we had a meeting and Rose F pointed out that it's pretty problematic for us to assert that anything is fully "accessible" since that's something that depends on the user and context. So the a11y issues thing is just supposed to print on its own beneath the checklist. 
Right now it has some pesky brackets `{}` that are creeping in there from I dont know where, but that's a separate problem...

Reply:
Oh okay I see! Is it a thing we're letting slide at the moment in the interest of moving forward with modules, or should that be a task we work on in the near future?

Reply:
I think we should fix it ASAP, but either way it's a problem in the macros_wrapper file, not anything in this actual module. So it would be a separate PR from this one.

Reply:
Oh okay that makes sense!